### PR TITLE
feat: add name when receiving Market Participant Created event

### PR DESCRIPTION
**Pull request by Mech0z**
## Description

MarketParticipantCreated event did not save Name when creating an MP, this addresses this.

Also renames MarketParticipantB2CActorIdEndpoint to MarketParticipantB2CActorIdChangedEndpoint

**note**
source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Application/MarketParticipants/Handlers/MarketParticipantPersister.cs will save empty string, but this will be removed when #1730 is done. 

## References

* #1726

### CODE REVIEW

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Application/MarketParticipants/Handlers/MarketParticipantCreatedCommandHandlerTests.cs [hunk 1]
```diff
@@ -42,6 +42,7 @@ public class MarketParticipantCreatedCommandHandlerTests
             var marketParticipantCreatedEvent = new MarketParticipantCreatedCommand(
                 ActorId: Guid.NewGuid(),
                 "mp123",
+                "name",
```
Comment (prtandrup) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Application/MarketParticipants/Handlers/MarketParticipantCreatedCommandHandlerTests.cs:45]: nit: add name to the assert list below

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Domain/MarketParticipants/MarketParticipantTests.cs [hunk 2]
```diff
@@ -28,11 +28,12 @@ public class MarketParticipantTests
         public void Create_ReturnsMarketParticipant(
             Guid actorId,
             string marketParticipantId,
+            string marketParticipantName,
             MarketParticipantStatus status,
             MarketParticipantRole businessProcessRole)
         {
             var actual =
-                MarketParticipant.Create(actorId, marketParticipantId, status, businessProcessRole);
+                MarketParticipant.Create(actorId, marketParticipantId, marketParticipantName, status, businessProcessRole);
```
Comment (prtandrup) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Domain/MarketParticipants/MarketParticipantTests.cs:36]: nit: Extend assert list with name

**Review by prtandrup — APPROVED** [review]
Approved with a few minor comments. 
I think the `string.Empty` is all right for the "Persister-way". Once turned off, it will become clear which market participants that were updated by the deprecated way, and we'll handle it from there.

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Domain/MarketParticipants/MarketParticipant.cs [hunk 3]
```diff
@@ -122,6 +133,11 @@ public void UpdateStatus(MarketParticipantStatus status)
             Status = status;
         }
 
+        public void UpdateName(string name)
```
[the remark this answers is not in the document]
Comment (Mech0z) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Domain/MarketParticipants/MarketParticipant.cs:136]: Removed, adding it to next PR

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Application/MarketParticipants/Handlers/MarketParticipantCreatedCommandHandlerTests.cs (hunk 1 above)
Comment (Mech0z) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Application/MarketParticipants/Handlers/MarketParticipantCreatedCommandHandlerTests.cs:45]: added

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Domain/MarketParticipants/MarketParticipantTests.cs (hunk 2 above)
Comment (Mech0z) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Domain/MarketParticipants/MarketParticipantTests.cs:36]: added

**Pull request merged**